### PR TITLE
OGW: compute T tensor from topography

### LINKED ISSUE #1430: OGW: compute orographic tensor directly from elevation
Currently, the orographic tensor (which contains all topography related information including amplitude, variance, orientation, and anisotropy) is obtained from a restart file of GFDL model. We need to directly compute it from the known elevation.

### PR DESCRIPTION
**Pull request by jiahe23**
co-authored by: @szy21 

This PR calculates T tensor following Steve Garner's method. Need to confirm (noted in the comments):
- can we directly remove `bfscale` in the calculation;
- is there additional smoothing applied? or how does the interpolation handle smoothing?

closes #1430

### CODE REVIEW

File: src/parameterizations/gravity_wave_drag/orographic_gravity_wave_helper.jl [hunk 1]
```diff
@@ -3,12 +3,186 @@ using Interpolations
...
+            wts =
+                reshape(
+                    repeat(cosd.(lat[jrange]), length(irange)),
+                    length(jrange),
+                    :,
+                )' ./ (arc .+ eps(FT)) .* (
+                    FT(0.42) .+ FT(0.50) .* cos.(pi * arc1) .+
+                    FT(0.08) .* cos.(FT(2) * pi * arc1)
+                )
+            if i != irange[1] && j != jrange[1]
```
Comment (szy21) [inline src/parameterizations/gravity_wave_drag/orographic_gravity_wave_helper.jl:147]: You probably don't need this line anymore. Or if you still need it, should it be `if i != irange[1]-1 && j != jrange[1]-1`?

File: src/parameterizations/gravity_wave_drag/orographic_gravity_wave_helper.jl [hunk 2]
```diff
@@ -51,7 +225,60 @@ function calc_hmax_latlon(elev, lon, lat, earth_radius)
...
+    # compute hmax and hmin
+    hmax = calc_hmax_latlon(elev, lon, lat, earth_radius)
+
+    # compute χ
+    χ = calc_velocity_potential(elev, lon, lat, earth_radius)
+
+    # compute orographic tensor (t11, t21, t12, t22)
+    t11, t21, t12, t22 = calc_orographic_tensor(elev, χ, lon, lat, earth_radius)
+
+    topo_ll = (; hmax = hmax, hmin = FT(0.1) .* hmax, t11, t12, t21, t22)
```
Comment (szy21) [inline src/parameterizations/gravity_wave_drag/orographic_gravity_wave_helper.jl:248]: [Here](https://github.com/NOAA-GFDL/atmos_phys/blob/def83fee4a751e0d0762c1ae5159e13042129116/atmos_param/topo_drag/topo_drag.F90#L863) is the relationship between hmax and hpos (which we call hmax here) in the gfdl code. Would it be possible to add it here?

**Review by szy21 — APPROVED** [review]
Thanks for wrapping up the gravity wave!

File: src/parameterizations/gravity_wave_drag/orographic_gravity_wave_helper.jl (hunk 2 above)
Comment (jiahe23) [reply src/parameterizations/gravity_wave_drag/orographic_gravity_wave_helper.jl]: I changed the function name to calc_hpoz_latlon and computed hmax from hpoz

**Pull request merged**